rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/cluster.desired_nodes/10_basic.yml

```diff
@@ -0,0 +1,364 @@
+---
+  setup:
+    - skip:
+        version: " - 8.0.99"
+        reason: "API added in in 8.1.0"
+    - do:
+        desired_nodes.delete_desired_nodes: {}
+---
+"Test update desired nodes":
+  - do:
+      desired_nodes.update_desired_nodes:
+        history_id: "test"
+        version: 1
+        body:
+            nodes:
+              - { settings: { "node.external_id": "instance-000187" }, processors: 8, memory: "64gb", storage: "128gb", version: "7.16.0" }
```

Comment:
>processors should likely be 1, otherwise it will not pass node.processors validation?

In this case we're not validating the `node.processors` setting, so it should be fine. There's a test that checks the setting override `Test some settings can be overridden`.

> version going backwards seems like something we should reject up front?

That sounds sensible, I'm not sure if there's a way to inject the current version in these tests instead of a hardcoded value so the test don't fail after we move to another version.

> Can we also tear down the desired nodes to avoid them having an effect on follow-on tests?

We already do that in the setup section, did you mean something else?

Reply:
Yeah, just want it in a teardown: section instead.